apps/workers/bigquery/worker.go

```diff
@@ -0,0 +1,279 @@
+	w := &worker{
+		Meta:    task.NewMeta(),
+		options: *o,
+	}
+	err := uri.Unmarshal(info, w)
+	if err != nil {
+		return task.InvalidWorker(err.Error())
+	}
+
+	// verify options
+	w.delete = len(w.DeleteMap) > 0
+	if w.delete && w.Truncate {
+		return task.InvalidWorker("truncate and delete options must be selected independently")
+	}
+
+	//TODO: verify destination is required for all loading, but only for reading in the dest_path is empty
+	return w
+}
+
+func (w *worker) DoTask(ctx context.Context) (task.Result, string) {
+	opts := make([]option.ClientOption, 0)
+	if w.BqAuth != "" {
+		opts = append(opts, option.WithCredentialsFile(w.BqAuth))
+	}
+	client, err := bigquery.NewClient(ctx, w.Project, opts...)
+	if err != nil {
+		return task.Failf("bigquery client init %s", err)
+	}
+	defer client.Close()
+
+	var format bigquery.DataFormat
+	switch filepath.Ext(w.File) {
+	case ".sql":
+		f, err := file.NewReader(w.File, &w.Fopts)
+		if err != nil {
+			return task.Failf("read error: %v %v", w.File, err)
+		}
+		b, _ := io.ReadAll(f)
+		query := string(b)
+		if isGCSExport(w.DestPath) {
+			query = addExportToGCS(b, w.DestPath, bigquery.CSV)
+		} else if w.DestPath != "" {
+			w.writeToFile = true
+		}
+		return w.Query(ctx, client, query)
+	case ".json":
+		format = bigquery.JSON
+		return w.Load(ctx, client, format)
+	case ".csv":
+		format = bigquery.CSV
+		return w.Load(ctx, client, format)
+	}
+
+	return task.Failf("unsupported file format %v, expected:sql|json|csv", filepath.Ext(w.File))
+}
+
+// isGCSExport checks the url path to see it is meant to be exported to GCS through BigQuery
+// - starts with gs://
+// - must contain a single asterisk * in the filename
+func isGCSExport(s string) bool {
+	if !strings.HasPrefix(s, "gs://") {
+		return false
+	}
+	d, f := filepath.Split(s)
+	return !strings.Contains(d, "*") && strings.Count(f, "*") == 1
+}
+
+func (w *worker) Load(ctx context.Context, client *bigquery.Client, format bigquery.DataFormat) (task.Result, string) {
+	var loader *bigquery.Loader
+	if w.FromGCS { // load from Google Cloud Storage
+		gcsRef := bigquery.NewGCSReference(w.File)
+		gcsRef.SourceFormat = format
+		loader = w.DestTable.BqTable(client).LoaderFrom(gcsRef)
+	} else { // load from file reader
+		r, err := file.NewReader(w.File, &w.Fopts)
+		if err != nil {
+			return task.Failf("problem with file: %s", err)
+		}
+		bqRef := bigquery.NewReaderSource(r)
+		bqRef.SourceFormat = format
+		bqRef.MaxBadRecords = 1
+		loader = w.DestTable.BqTable(client).LoaderFrom(bqRef)
+	}
+
+	loader.WriteDisposition = bigquery.WriteAppend
+	if len(w.DeleteMap) > 0 {
+		q := delStatement(w.DeleteMap, w.DestTable)
+		j, err := client.Query(q).Run(ctx)
+		if err != nil {
+			return task.Failf("delete statement: %s", err)
+		}
+		status, err := j.Wait(ctx)
+		if err != nil {
+			return task.Failf("delete wait: %s", err)
+		}
+		if status.Err() != nil {
+			return task.Failf("delete: %s", err)
```

Comment:
if status.Err() is not nill, shouldn't you be returning `return task.Failf("delete: %s", status.Err())`